File: examples/python/send_bulk_sms.py
"""Send several SMS in a single request with send_bulk.

Run:  PUSHFY_API_TOKEN=... python3 send_bulk_sms.py

For very large lists (tens of thousands of recipients) split into chunks —
see batch_send.py.
"""

import os

from pushfy import Pushfy


def main():
    pushfy = Pushfy(api_token=os.environ["PUSHFY_API_TOKEN"])

    # Each message carries its own unique ext_id so you can correlate the
    # per-recipient delivery status later.
    messages = [
        {"to": "5511999990001", "text": "Hi Ana",   "ext_id": "promo-0001"},
        {"to": "5511999990002", "text": "Hi Bruno", "ext_id": "promo-0002"},
        {"to": "5511999990003", "text": "Hi Carla", "ext_id": "promo-0003"},
    ]

    result = pushfy.sms.send_bulk(messages)
    print("Accepted %d messages:" % len(messages))
    print(result)


if __name__ == "__main__":
    main()

File: examples/php/send-bulk-sms.php
<?php

declare(strict_types=1);

// Send many SMS in a single API request (one call, many recipients).
//   PUSHFY_API_TOKEN=... php send-bulk-sms.php
//
// Use sendBulk() when you already have the full list in memory. For very
// large lists that must be split into fixed-size requests, see batch-send.php.

require __DIR__ . '/vendor/autoload.php';

use Pushfy\Pushfy;
use Pushfy\Exception\PushfyException;

$pushfy = new Pushfy([
    'apiToken' => getenv('PUSHFY_API_TOKEN') ?: '',
]);

// Each recipient carries its own unique extId (your idempotency key).
$messages = [
    ['to' => '5511999999999', 'text' => 'Hi Ana, your code is 1234',   'extId' => 'otp-ana-001'],
    ['to' => '5511999999999', 'text' => 'Hi Bruno, your code is 5678', 'extId' => 'otp-bruno-002'],
    ['to' => '5511999999999', 'text' => 'Hi Célia, your code is 9012', 'extId' => 'otp-celia-003'],
];

try {
    $result = $pushfy->sms->sendBulk($messages);

    echo "Accepted ", count($messages), " messages:\n";
    echo json_encode($result, JSON_PRETTY_PRINT | JSON_UNESCAPED_UNICODE), "\n";
} catch (PushfyException $e) {
    fwrite(STDERR, sprintf(
        "Bulk send failed: status=%d code=%s\n",
        $e->getStatus(),
        (string) $e->getErrorCode()
    ));
    exit(1);
}
